test/e2e_node/e2e_remote.go

```diff
@@ -0,0 +1,178 @@
+		glog.Fatalf("Failed to locate test binary %s", ginkgoTest)
+	}
+	defer os.Remove(ginkgoTest)
+
+	// Make sure we can find the newly built binaries
+	buildOutputDir, err := getK8sBuildOutputDir()
+	if err != nil {
+		glog.Fatalf("Failed to locate kubernetes build output directory %v", err)
+	}
+	kubelet := filepath.Join(buildOutputDir, "kubelet")
+	if _, err := os.Stat(kubelet); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate binary %s", kubelet)
+	}
+	apiserver := filepath.Join(buildOutputDir, "kube-apiserver")
+	if _, err := os.Stat(apiserver); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate binary %s", apiserver)
+	}
+
+	glog.Infof("Building archive...")
+	tardir, err := ioutil.TempDir("", "node-e2e-archive")
+	if err != nil {
+		glog.Fatalf("Failed to create temporary directory %v.", err)
+	}
+	defer os.RemoveAll(tardir)
+
+	// Copy binaries
+	out, err = exec.Command("cp", ginkgoTest, filepath.Join(tardir, "e2e_node.test")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to copy e2e_node.test %v.", err)
+	}
+	out, err = exec.Command("cp", kubelet, filepath.Join(tardir, "kubelet")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to copy kubelet %v.", err)
+	}
+	out, err = exec.Command("cp", apiserver, filepath.Join(tardir, "kube-apiserver")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to copy kube-apiserver %v.", err)
+	}
+
+	// Build the tar
+	out, err = exec.Command("tar", "-zcvf", "e2e_node_test.tar.gz", "-C", tardir, ".").CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to build tar %v.  Output:\n%s", err, out)
+	}
+
+	dir, err := os.Getwd()
+	if err != nil {
+		glog.Fatalf("Failed to get working directory %v.", err)
+	}
+	return filepath.Join(dir, "e2e_node_test.tar.gz")
+}
+
+// RunRemote copies the archive file to a /tmp file on host, unpacks it, and runs the e2e_node.test
+func RunRemote(archive string, host string) (string, error) {
+	// Create the temp staging directory
+	tmp := fmt.Sprintf("/tmp/gcloud-e2e-%d", rand.Int31())
+	_, err := runSshCommand("ssh", host, "--", "mkdir", "-p", tmp)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		output, err := runSshCommand("ssh", host, "--", "rm", "-rf", tmp)
+		if err != nil {
+			glog.Errorf("Failed to cleanup tmp directory %s on host %v.  Output:\n%s", tmp, err, output)
+		}
+	}()
+
+	// Copy the archive to the staging directory
+	_, err = runSshCommand("scp", archive, fmt.Sprintf("%s:%s/", host, tmp))
+	if err != nil {
+		return "", err
+	}
+
+	// Kill any running node processes
+	cmd := getSshCommand(" ; ",
+		"sudo pkill kubelet",
+		"sudo pkill kube-apiserver",
+		"sudo pkill etcd")
+	runSshCommand("ssh", host, "--", "sh", "-c", cmd)
+
+	// Extract the archive and run the tests
+	cmd = getSshCommand(" && ",
+		fmt.Sprintf("cd %s", tmp),
+		"tar -xzvf ./e2e_node_test.tar.gz",
+		"./e2e_node.test --alsologtostderr --v 2 --build-services=false --node-name `hostname`")
```

Comment:
nit: `-logtostderr` is better since it avoids accumulating log files on the test node.